common/logging/errorLogging.go

```diff
@@ -0,0 +1,33 @@
+package logging
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	elrondErrors "github.com/ElrondNetwork/elrond-go/errors"
+)
+
+// LogErrAsWarnExceptAsDebugIfClosingError logs an error
+func LogErrAsWarnExceptAsDebugIfClosingError(logInstance logger.Logger, err error, message string, args ...interface{}) {
+	logErrAsLevelExceptAsDebugIfClosingError(logInstance, logger.LogWarning, err, message, args...)
+}
+
+// LogErrAsErrorExceptAsDebugIfClosingError logs an error
+func LogErrAsErrorExceptAsDebugIfClosingError(logInstance logger.Logger, err error, message string, args ...interface{}) {
+	logErrAsLevelExceptAsDebugIfClosingError(logInstance, logger.LogError, err, message, args...)
+}
+
+func logErrAsLevelExceptAsDebugIfClosingError(logInstance logger.Logger, logLevel logger.LogLevel, err error, message string, args ...interface{}) {
```

Comment:
I had a similar function in one of my earlier PR. I would switch this to:
```GO
func logErrAsLevelExceptAsDebugIfClosingError(logInstance logger.Logger, logLevel logger.LogLevel, err error, message string, args ...interface{}) {
    if check.IfNil(logInstance) {
        return
    }

    if elrondErrors.IsClosingError(err) {
        logLevel = logger.LogDebug
    }

    logInstance.Log(logLevel, message, argsWithError...)
}
```

This is a simplified version that has the following advantage: 
we can log the error as we want, because a call can be made like this:
```GO
logErrAsLevelExceptAsDebugIfClosingError(log, logger.LogError, err, "message", "shard", shardID, "error", err)
```
or
```GO
logErrAsLevelExceptAsDebugIfClosingError(log, logger.LogError, err, "message", "error", err, "shard", shardID)
```

Reply:
Fixed.